BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/ReflectionCredentialEnvelope.cs

```diff
@@ -0,0 +1,205 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Authentication
+{
+    using System;
+    using System.Linq.Expressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// This is an implementation of <see cref="CredentialEnvelope"/> that uses reflection to interact with the Azure.Core library.
+    /// </summary>
+    /// <remarks>
+    /// Our SDK currently targets net452, net46, and netstandard2.0.
+    /// Azure.Core.TokenCredential is only available for netstandard2.0.
+    /// </remarks>
+    internal class ReflectionCredentialEnvelope : CredentialEnvelope
```

Comment:
if this reflection based approach works good enough, is there any reason then to depend on azure.core in netstandard case?